src/recipe/recipe.controller.ts

```diff
@@ -1,14 +1,20 @@
-import { Controller, Delete, Get, NotFoundException, Param, Query, Req, UseGuards } from '@nestjs/common';
+import { Body, Controller, Delete, Get, NotFoundException, Param, Post, Query, Req, UseGuards } from '@nestjs/common';
 import { OwnerGuard } from '../auth/guards/owner.guard';
 import { PaginatedQueryDto } from '../dto/queries.dto';
-import { RecipeDto, RecipeIdDto } from './dto/recipe.dto';
+import { RecipeDto, RecipeContentDto, RecipeIdDto } from './dto/recipe.dto';
 
 import { RecipeService } from './recipe.service';
 
 @Controller('recipe')
 export class RecipeController {
   constructor(private recipeService: RecipeService) {}
 
+  @Post('/')
+  createRecipe(@Body() recipeContents: RecipeContentDto, @Req() req: any): Promise<RecipeDto> {
```

Comment:
@Kirosana15 You could create a decorator to extract `user` from `request`

```ts
import { createParamDecorator, ExecutionContext } from '@nestjs/common';
import { User } from 'user.dto'; // I don't know path :P 

export const AuthorizedUser = createParamDecorator(
  (data, ctx: ExecutionContext): User => {
    const req = ctx.switchToHttp().getRequest();
    return req.user;
  },
);
```

and use it like this

```ts
createRecipe(@Body() recipeContents: RecipeContentDto, @AuthorizedUser() user: User)
```